service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/base/SharedNotifyContentProvider.java

```diff
@@ -2,47 +2,59 @@
 import static org.apache.commons.lang3.ObjectUtils.defaultIfNull;
+import static uk.gov.hmcts.reform.fpl.enums.YesNo.NO;
+import static uk.gov.hmcts.reform.fpl.enums.YesNo.YES;
 import static uk.gov.hmcts.reform.fpl.utils.EmailNotificationHelper.formatCaseUrl;
 import static uk.gov.hmcts.reform.fpl.utils.PeopleInCaseHelper.getFirstRespondentLastName;
 
-public abstract class CasePersonalisedContentProvider extends AbstractEmailContentProvider {
-    protected CasePersonalisedContentProvider(String uiBaseUrl, ObjectMapper mapper) {
+public abstract class SharedNotifyContentProvider extends AbstractEmailContentProvider {
+    protected SharedNotifyContentProvider(String uiBaseUrl, ObjectMapper mapper) {
         super(uiBaseUrl, mapper);
     }
 
-    protected ImmutableMap.Builder<String, Object> getCasePersonalisationBuilder(Long caseId, CaseData caseData) {
-        List<String> ordersAndDirections = buildOrdersAndDirections(caseData.getOrders());
+    protected <T extends SharedNotifyTemplate> T buildNotifyTemplate(T template,
+                                                                     Long caseId,
+                                                                     Orders orders,
+                                                                     Hearing hearing,
+                                                                     List<Element<Respondent>> respondents1) {
+        List<String> ordersAndDirections = buildOrdersAndDirections(orders);
 
-        Optional<String> timeFrame = Optional.ofNullable(caseData.getHearing())
+        Optional<String> timeFrame = Optional.ofNullable(hearing)
             .map(Hearing::getTimeFrame)
             .filter(StringUtils::isNotBlank);
 
-        return ImmutableMap.<String, Object>builder()
-            .put("ordersAndDirections", !ordersAndDirections.isEmpty() ? ordersAndDirections : "")
-            .put("dataPresent", !ordersAndDirections.isEmpty() ? "Yes" : "No")
-            .put("fullStop", !ordersAndDirections.isEmpty() ? "No" : "Yes")
-            .put("timeFramePresent", timeFrame.isPresent() ? "Yes" : "No")
-            .put("timeFrameValue", uncapitalize(timeFrame.orElse("")))
-            .put("urgentHearing", timeFrame.isPresent() && timeFrame.get().equals("Same day") ? "Yes" : "No")
-            .put("nonUrgentHearing", timeFrame.isPresent() && !timeFrame.get().equals("Same day") ? "Yes" : "No")
-            .put("firstRespondentName", getFirstRespondentLastName(caseData.getRespondents1()))
-            .put("reference", String.valueOf(caseId))
-            .put("caseUrl", formatCaseUrl(uiBaseUrl, caseId));
+        // What if we just add the list no matter if it is empty or not,
+        // surely the parts below determine if it is going to be shown or not
+        template.setOrdersAndDirections(!ordersAndDirections.isEmpty() ? ordersAndDirections : List.of(""));
+        template.setDataPresent(!ordersAndDirections.isEmpty() ? YES.getValue() : NO.getValue());
```

Comment:
Did you investigate adding JsonProperty tags to the YES and NO enums so that you can directly pass them instead of getting the value attached?

Reply:
Or did it just end up breaking things in other places?

Reply:
Yes I did and reverted :/ It impacted the placementController and was not keen on adding out of scope changes.